helper-cli/src/main/kotlin/commands/SetDependencyRepresentationCommand.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.helper.commands
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.UsageError
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.default
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.enum
+import com.github.ajalt.clikt.parameters.types.file
+
+import org.ossreviewtoolkit.analyzer.AnalyzerResultBuilder
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.model.utils.DependencyGraphConverter
+import org.ossreviewtoolkit.model.writeValue
+import org.ossreviewtoolkit.utils.expandTilde
+
+/**
+ * A command to convert the dependency representation in an ORT result file between the different supported formats.
+ *
+ * The input file can contain the serialized form of either a full [OrtResult], an [AnalyzerResult], or a single
+ * [ProjectAnalyzerResult].
+ */
+class SetDependencyRepresentationCommand : CliktCommand(
+    help = "Convert an ORT result from the dependency tree representation to the dependency graph format."
+) {
+    /**
+     * An enumeration class representing the different supported formats to represent dependencies in an ORT result.
+     * It is used to define the target format that should be set by this command.
+     */
+    enum class TargetFormat {
+        /** The dependency graph format. */
+        GRAPH {
```

Comment:
just out of interrest: Are there plans to implement conversion from `Graph` to `Scope`?

Reply:
Not concrete yet. What we have discussed was mainly to see the diffs for failing tests in the more readable tree format; but this could be achieved via a custom matcher. However, the conversion to the classic format is already implemented, so if there is a concrete use case, why not.

Reply:
I believe debugging could be a use case, e.g. making it more readable to easier grasp it.